client/checkout/utils/upe.js

```diff
@@ -25,42 +30,23 @@ export const getTerms = ( paymentMethodsConfig, value = 'always' ) => {
 };
 
 /**
- * Finds selected payment gateway and returns matching Stripe payment method for gateway.
+ * Returns Stripe payment method (e.g. card, bancontact ) for selected payment gateway.
  *
- * @return {string} Stripe payment method type
+ * @return {string} Payment method name
  */
 export const getSelectedUPEGatewayPaymentMethod = () => {
-	const paymentMethodsConfig = getUPEConfig( 'paymentMethodsConfig' );
-	const gatewayCardId = getUPEConfig( 'gatewayId' );
-	let selectedGatewayId = null;
-
-	// Handle payment method selection on the Checkout page or Add Payment Method page where class names differ.
-	const radio = document.querySelector(
-		'li.wc_payment_method input.input-radio:checked, li.woocommerce-PaymentMethod input.input-radio:checked'
+	const selectedGateway = document.querySelector(
+		`.wcpay-upe-form[checked]`
 	);
-	if ( radio !== null ) {
-		selectedGatewayId = radio.id;
-	}
-
-	if ( selectedGatewayId === 'payment_method_woocommerce_payments' ) {
-		selectedGatewayId = 'payment_method_woocommerce_payments_card';
-	}
-
-	let selectedPaymentMethod = null;
-
-	for ( const paymentMethodType in paymentMethodsConfig ) {
-		if (
-			`payment_method_${ gatewayCardId }_${ paymentMethodType }` ===
-			selectedGatewayId
-		) {
-			selectedPaymentMethod = paymentMethodType;
-			break;
-		}
-	}
-
-	return selectedPaymentMethod;
+	return selectedGateway?.dataset.paymentMethodType || null;
```

Comment:
`payment_method_selected` handler in `event-handlers.js` should do everything for us and we just need to return the gateway that is checked.